src/tribler-core/tribler_core/components/restapi/restapi_component.py

```diff
@@ -126,6 +126,9 @@ def report_callback(reported_error: ReportedError):
             self._events_endpoint.on_tribler_exception(reported_error)
 
         self._core_exception_handler.report_callback = report_callback
+        # Reraise the unreported error, if there is one
+        if self._core_exception_handler.unreported_error:
+            report_callback(self._core_exception_handler.unreported_error)
```

Comment:
**Note** it is better to set `unreported_error` to None to indicate that the error has been reported.

```suggestion
            report_callback(self._core_exception_handler.unreported_error)
            self._core_exception_handler.unreported_error = None
```